cmd/minikube/cmd/start.go

```diff
@@ -141,6 +145,15 @@ func runStart(cmd *cobra.Command, args []string) {
 		glog.Errorln("Error starting the VM: ", err)
 		os.Exit(1)
 	}
+
+	// Set Proxy to as Shell Env
+	if viper.IsSet("http-proxy") || viper.IsSet("https-proxy") {
```

Comment:
> Ok, we might change the retry then. I don't see much use in this anyway. Unless we are covering for a error condition which occurs sporadically.

It's required because many time cert creation for ssh need time and that time it do the retry (Which I observed some time back).

> Otherwise, we at least need to move the functionality of registration into a public method in cluster.

That I can take a look but I would make this part of change in https://github.com/minishift/minishift/issues/318